lib/lightning_web/live/components/credentials.ex

```diff
@@ -357,7 +347,10 @@ defmodule LightningWeb.Components.Credentials do
               phx-click={show_modal(target)}
               disabled={@disabled}
             >
-              <%= name %>
+              <%= name %><span
+                :if={Map.get(option, :badge)}
+                class="ml-2 inline-flex items-center rounded-md bg-gray-50 px-1.5 py-0.5 text-xs font-medium text-gray-600 ring-1 ring-inset ring-gray-500/10"
+              ><%= Map.get(option, :badge) %></span>
             </a>
```

Comment:
I think this interface is kinda confused, this is best encapsulated using slots.

The `options` assign for creating n number of a tags is exactly where slots are best. 

Also I see the button is only rendered in one place, with one set of options. I don't see the need for it to be so generic vs a composed function component that reuses other function components (if any).